### PR TITLE
task feed ranking added, worker side

### PR DESCRIPTION
**Pull request by dmorina**
left outer join btw :P

### CODE REVIEW

File: crowdsourcing/viewsets/project.py [hunk 1]
```diff
@@ -63,11 +63,23 @@ def update_project(self, request, pk=None):
...
+                  INNER JOIN crowdsourcing_requester r ON p.owner_id = r.id
+                  INNER JOIN crowdsourcing_userprofile u ON r.profile_id = u.id
+                  LEFT OUTER JOIN crowdsourcing_workerrequesterrating w ON u.id = w.target_id
+                    AND w.type='worker' AND w.origin_id=%s
+                  LEFT OUTER JOIN (SELECT target_id,  AVG(weight) AS average_rating  from
+                    crowdsourcing_workerrequesterrating where type='worker' GROUP BY target_id) avg
+                    ON avg.target_id = u.id) calc order by relevant_rating desc
+            ''', params=[request.user.userprofile.id])
+            projects_serialized = ProjectSerializer(projects, many=True)
             return Response(projects_serialized.data)
```
Comment (dmorina) [inline crowdsourcing/viewsets/project.py]: this has changed only, the rest is reformatting

**Comment by dmorina** [comment]
@aginzberg  have a look

File: crowdsourcing/viewsets/project.py [hunk 2]
```diff
@@ -63,11 +63,23 @@ def update_project(self, request, pk=None):
...
-            projects = models.Project.objects.prefetch_related('owner__profile__rating_target')
-            projects_filtered = projects.filter(owner__profile__rating_target__origin=request.user.userprofile,
-                                                owner__profile__rating_target__type='worker')
-            projects_ordered = projects_filtered.order_by('-owner__profile__rating_target__weight')
-            projects_serialized = ProjectSerializer(projects_ordered, many=True)
+            projects = Project.objects.raw('''
+                select id, name, description, created_timestamp, last_updated, owner_id, case when weight is null
+                and average_rating is not null then average_rating
+                when weight is null and average_rating is null then 1.99
+                when weight is not null and average_rating is null then weight
```
Comment (aginzberg) [inline crowdsourcing/viewsets/project.py]: Do we need this?

File: crowdsourcing/viewsets/project.py (hunk 1 above)
Comment (aginzberg) [inline crowdsourcing/viewsets/project.py]: This still returns projects you have created as a requester.

File: crowdsourcing/viewsets/project.py (hunk 2 above)
Comment (dmorina) [inline crowdsourcing/viewsets/project.py]: Not necessarily if we count ourselves in average

File: crowdsourcing/viewsets/project.py (hunk 1 above)
Comment (dmorina) [inline crowdsourcing/viewsets/project.py]: Yes, we can filter out those too

Comment (aginzberg) [inline crowdsourcing/viewsets/project.py]: lgtm. will filter projects i created as requester after merge.

**Pull request merged**